src/SmartComponents/RunTaskModal/ParameterMultiSelect.js

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from 'react';
+import {
+  Select,
+  SelectOption,
+  SelectList,
+  MenuToggle,
+  Badge,
+  HelperText,
+  HelperTextItem,
+  FormGroup,
+  FormHelperText,
+} from '@patternfly/react-core';
+import propTypes from 'prop-types';
+
+const ParameterMultiSelect = ({
+  parameter,
+  updateParameter,
+  customDescription,
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [parameterValue, setParameterValue] = useState([]);
+
+  const onToggleClick = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const onSelect = (_event, value) => {
+    const newParameterValue = parameterValue.includes(value)
+      ? parameterValue.filter((val) => val !== value)
+      : [...parameterValue, value];
+    setParameterValue(newParameterValue);
+    updateParameter(parameter, newParameterValue.join(',') || 'None');
+  };
+
+  const toggle = (toggleRef) => (
+    <MenuToggle ref={toggleRef} onClick={onToggleClick} isExpanded={isOpen}>
+      Select {parameter.title || parameter.key}
+      {parameterValue.length > 0 && (
+        <Badge isRead>{parameterValue.length}</Badge>
```

Comment:
We could use `badge` prop instead for the selection count:

https://www.patternfly.org/components/menus/menu-toggle#with-a-badge

This will also keep us consistent with PF design guidelines.

Reply:
Ok, I'll have a look.  But TBH I just used the example code from Patternfly's checkbox-select component and modified it for Tasks: https://www.patternfly.org/components/menus/select#checkbox-select

Reply:
Done. Yep, using a prop seems a cleaner way to do the badge.